### PR TITLE
Module federation fix & update

### PR DESCRIPTION
**Pull request by JulianWielga**

### CODE REVIEW

File: ui/client/webpack.config.js [hunk 1]
```diff
@@ -171,7 +171,6 @@ module.exports = {
       base: isProd ? "__publicPath__/static/" : "/",
       filename: "main.html",
       favicon: "assets/img/favicon.png",
-      template: "index_template_no_doctype.ejs",
```
Comment (mproch) [inline ui/client/webpack.config.js:155]: let's remove also the file itself?

Comment (JulianWielga) [reply ui/client/webpack.config.js:155]: removed with styles fixed

File: ui/client/federation.config.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "nussknackerUi",
+  "exposes": {
```
Comment (arkadius) [inline ui/client/federation.config.json:3]: Can you add some links to documentation of this file content in documentation?

File: ui/client/config.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/*
+  We want to be able to serve NK under different (in particular not empty) paths (e.g. via nginx configuration) without rebuilding frontends
+  The easiest way would be to use relative urls, but because we don't use hash router, but rely on different paths, it's more difficult
+  Currently we replace __publicPath__ string in main.html during serving with configured path, and we use "auto" publicPath from webpack
+  via global variable (__webpack_public_path__).
+  __webpack_public_path__ is also used when this module is imported in external app (WebpackModuleFederationPlugin) because it contains
```
Comment (arkadius) [inline ui/client/config.ts:6]: Look like important entry to documentation about how to use federated modules

File: ui/client/components/DragArea.js [hunk 4]
```diff
@@ -3,12 +3,13 @@ import {DragDropContext} from "react-dnd"
 
 import HTML5Backend from "react-dnd-html5-backend"
 
+//TODO: this looks wierd, consider remove
 class DragArea extends React.Component {
   render() {
     return (
-      <div>
+      <>
```
Comment (arkadius) [inline ui/client/components/DragArea.js:10]: what? how? why? ;D

File: ui/client/package.json [hunk 5]
```diff
@@ -29,7 +29,8 @@
     "test:e2e:linux:update": "docker run --add-host=host.docker.internal:host-gateway -it -v $PWD:/client -w /client -e CYPRESS_SNAPSHOT_UPDATE=true cypress/included:6.8.0 --config baseUrl=http://host.docker.internal:3000",
     "test:e2e:docker:update": "start-server-and-test start:backend-docker http-get://localhost:3000/static/main.html test:e2e:linux:update",
     "coverage": "npx merge-cypress-jest-coverage",
-    "commit": "lint-staged"
+    "commit": "lint-staged",
+    "make-types": "make-federated-types --outputDir ./.federated-types"
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply ui/client/package.json:33]: used only by Webpack to do most things automagically, trying to leave future devs free from thinking about this

[the remark this answers is not in the document]
Comment (JulianWielga) [reply ui/client/package.json:33]: used only by Webpack to do most things automagically, trying to leave future devs free from thinking about this

File: ui/client/webpack.config.js [hunk 6]
```diff
@@ -86,27 +82,37 @@ module.exports = {
...
     historyApiFallback: {
-      index: "/static/main.html",
+      index: "/main.html",
     },
     overlay: {errors: true, warnings: false},
     hot: true,
     host: "0.0.0.0",
     disableHostCheck: true,
+    headers: {
+      "Access-Control-Allow-Origin": "*",
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply ui/client/webpack.config.js:101]: look few lines below. this line is really not needed.

File: ui/client/components/DragArea.js (hunk 4 above)
Comment (JulianWielga) [reply ui/client/components/DragArea.js]: mistake, corrected

File: ui/client/config.ts (hunk 3 above)
Comment (JulianWielga) [reply ui/client/config.ts:6]: if you use Webpack, you know where it's docs lays. same with WebpackModuleFederationPlugin.  I think its enough.

**Review by arkadius — APPROVED** [review]
I'm not satisfied because of lack of documentation, but let's try and check how work without that will look like for next person...

**Pull request merged**